### DISCUSSION TITLE
How to generate test data for custom model?

### DISCUSSION THREAD

**Discussion by niclas-b** [microsoft/onnxruntime#11456] [Other Q&A]
Hi,

using the commando line I want to run the onnxruntime_perf_test.exe to optimize my custom onnx model's performance. It is not a model from the model zoo. What I don't find so far is how to generate dummy test data for input, like which data structure i need to use and in which file format i have to save it? Because the commando line is asking for this when I try to start the performance testing.
Best Regards
Niclas

**Comment by niclas-b** [comment]
Nevermind, I overlooked the corresponding argument to generate input tensor binding.

**Reply by yufenglee** [reply]
yes, option -I will generate inputs with random data.

**Comment by snnn** [comment]
https://github.com/microsoft/onnxruntime/pull/10909/ has an example. The PR adds a bunch of new python scripts. onnxruntime/test/python/generate_test_data.py is the main entry point.  onnxruntime/test/python/node/max_pool2d.py is an example of how to generate a TF then convert it to ONNX format and save the test data along with the model.
